$controllers returned always false

levi-vweb opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Bugfix when storing ps_facetedsearch template, for some reason the template edit form doesn't pass any "controllers" thus $controllers = Tools::getValue('controllers'); always returns false and shows the error message "You must select at least one page." , so if there are no controllers passed this "fix" will get the controllers list from the class attribute $supportedControllers

| Type?         | bug fix 

| Deprecations? |  no

| How to test?  | saving a faceted search template, without fix it (always) returns error message "You must select at least one page."

<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/PrestaShop/ps_facetedsearch/907)
<!-- Reviewable:end -->

Hlavtox commented:
@levi-vweb Hi, can you please provide steps to reproduce this issue? It should work correctly.

levi-vweb commented:
@Hlavtox the issue occurs when saving a ps_facetedsearch Filters templates  either via "edit" or via "add new template"

levi-vweb commented:
Hi @hlvatox hope you had a nice weekend, I forgot to mention it is happening on prestashop 8.1.0 with the latest ps_facetedsearch module I got from github. if that might help anything.
Well to be honest in my ps_facetedsearch module  there is no option for selecting any pages, the only things I can select are the categories and filters to apply. So I found some of the "controllers" it was asking for and just passed that array so it worked again. so it's mostly just a "hotfix" from my side.

I added some screenshots to imgur https://imgur.com/a/pvn7yYC so you can see my backoffice settings screens  for the module.
[https://i.imgur.com/yRaXivx.png?fb]<https://imgur.com/a/pvn7yYC>
Imgur: The magic of the Internet<https://imgur.com/a/pvn7yYC>
Discover the magic of the internet at Imgur, a community powered entertainment destination. Lift your spirits with funny jokes, trending memes, entertaining gifs, inspiring stories, viral videos, and so much more from users.
imgur.com

________________________________
From: Daniel Hlavacek ***@***.***>
Sent: Friday, 25 August 2023 17:31
To: PrestaShop/ps_facetedsearch ***@***.***>
Cc: Levi Brown ***@***.***>; Mention ***@***.***>
Subject: Re: [PrestaShop/ps_facetedsearch] $controllers returned always false (PR #907)


@levi-vweb<https://github.com/levi-vweb> Please, provide exact steps, because I implemented these things and everything is working fine I think.

—
Reply to this email directly, view it on GitHub<https://github.com/PrestaShop/ps_facetedsearch/pull/907#issuecomment-1693546463>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AYLL2PCV5FVA6IP5IJ6TZFDXXDALNANCNFSM6AAAAAA36EU32A>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

Hlavtox commented:
@levi-vweb Hi, something is broken in your module, is there something overriding the template? You are completely missing the “pages” section in the filter settings.

levi-vweb commented:
@Hlavtox hahaha yeah that would make sense, as I was wondering what that pages notifications was for in the first place, it's weird tho, since it happened at first with just a fresh install of the latest prestashop :/  then I replaced it with what I took from the github repo, but that didn't change anything, so I have no idea what might have broken it....
